R/enrichDAVID.R

```diff
@@ -82,86 +80,68 @@ enrichDAVID <- function(gene,
     david.res <- addList(david, gene, idType=idType,
                          listName="clusterProfiler",
                          listType="Gene")
-
-
+    
+    
     if (david.res$inDavid == 0) {
         stop("All id can not be mapped. Please check 'idType' parameter...")
     }
-
+    
     if (!missing(universe)) {
         david.res <- addList(david, universe, idType=idType,
                              listName="universe",
                              listType="Background")
     }
-
+    
     setAnnotationCategories(david, annotation)
     x <- getFunctionalAnnotationChart(david, threshold=1, count=minGSSize)
-
-    if (length(x@.Data) == 0) {
-        warning("No significant enrichment found...")
-        return(NULL)
+    
+    if (!is.na(maxGSSize) && !is.null(maxGSSize)) {
+        x <- x[x$Count <= maxGSSize, ]
     }
-
-    term <- x$Term
-    if (length(grep("~", term[1])) == 0) {
-        sep <- ":"
-    } else {
-        sep <- "~"
+    
+    if (nrow(x) == 0) {
+        warning("No enriched GO term found...")
+        return(NULL)
```

Comment:
not only GO terms can be tested.

Reply:
Totally agree, I was in the case of the GO annotation.